### ISSUE TITLE
List patterns: no error is emitted when Length is missing or inaccessible

### ISSUE THREAD

**Issue by SomeAnonDev** [dotnet/roslyn#59465]
**Version Used**: main (27 Jan 2022, 3b32fc36ec4c2a81457d90f965c89c5b6fc37082)

**Steps to Reproduce**:

```csharp
using System;

_ = new S() is [];

struct S
{
  public int this[Index i] => i.Value;
  // Introducing a private Length property won't change the result
  // private int Length => 42;
}
```

**Expected Behavior**: a compiler error is emitted saying that the ``Length`` property is required but missing/inaccessible.

**Actual Behavior**:
``error CS7038: Failed to emit module '_': Unable to determine specific cause of the failure.``

Relates to test plan https://github.com/dotnet/roslyn/issues/51289

**Comment by RikkiGibson** [comment]
fyi @alrz @jcouv

**Comment by jcouv** [comment]
Thanks @SomeAnonDev for reporting this. The fix was merged for 17.2p2.
